x-pack/legacy/plugins/siem/public/pages/detection_engine/rules/components/ml_job_select/index.tsx

```diff
@@ -77,26 +104,35 @@ export const MlJobSelect: React.FC<MlJobSelectProps> = ({ describedByIds = [], f
 
   const options = [placeholderOption, ...jobOptions];
 
+  const isJobRunning = useMemo(() => {
```

Comment:
In what situation would the job be null/undefined, here? Do we truly want to hide the warning in that case, or should we just return `isJobStarted(job?.jobState, job?.datafeedState)` instead?

Reply:
This is because of the initial placeholder value used in the `MlJobSelect`. If the job is not found in the list, it means the placeholder is selected, and thus we don't want to show the warning. The job being `null` makes this confusing, so I'll add a comment to clarify (tried to rework the logic, but just made it more confusing 😅).